### ISSUE TITLE
Suite quit unexpectedly - EHTOS Suite 1.5.1 - Mac 12.7.4 

### ISSUE THREAD

**Issue by spoke2570** [FrSkyRC/ETHOS-Feedback-Community#3616]
The Mac OS gives an "ETHOS quit unexpectedly" alert if the discs are ejected and the radio unplugged before quitting Suite. 
There are no "disc not properly ejected" errors. 
This does not happen if Suite is opened and closed without plugging in the radio. 
This does not happen if the radio is unplugged before quitting Suite.
Also this seems to have no effect on any of the functions while using Suite, so really minor issue.

**Comment by spoke2570** [comment]
I have not been able to make Suite 1.5.2 repeat this.

**Comment by Nicholas-Luoyi** [comment]
So it is fixed?

**Comment by spoke2570** [comment]
If Ethos For Mac didn't change from 1.5.1, then no:  just haven't made it happen in 1.5.2 yet. I did put some effort into recreating the fault. Leave this open for a week or 2 and if can make it happen again I'll grab a screen shot and a report from my OS.

**Comment by Nicholas-Luoyi** [comment]
I didn't change the code for this part. But the package environment has been updated. It would be good if you could help to keep taking care of the crash. Thanks

**Comment by spoke2570** [comment]
This is fixed. I took life off my USB port trying this morning.  Thank you!

**Comment by spoke2570** [comment]
I am attaching a screen shot of the expanded error notification, and a .txt of its content. 

This round I just looked at some settings turned on the popup window for fatal error and used the quit button provided by Mac in the Finder side bar.

I looked in Ethos Suite/Crashpad but found no report from Suite.

![Mac_OS_Problem_Report](https://github.com/FrSkyRC/ETHOS-Feedback-Community/assets/9330938/a7a66e95-5353-4647-a61d-05bb3a65f0f6)
[Mac_OS_Problem_Report.txt](https://github.com/FrSkyRC/ETHOS-Feedback-Community/files/14623558/Mac_OS_Problem_Report.txt)

**Comment by Nicholas-Luoyi** [comment]
Thanks!

**Comment by PeterGraz** [comment]
My experience with ETHOS Suite 1.5.2 is the following:
On my older Mac with OS X 10.15.7 the suite occasionally crashes also without plugging in the radio. The later versions 1.4.x of the ETHOS Suite crashed always after launching the app.
On my newer Mac with macOS 13.6.5 ETHOS Suite 1.5.2 crashes also occasionally, but I could not find a pattern so far. However, sometimes ETHOS Suite dismounts the discs "SD-Card" and "FLASH", but not the disc "Radio". The disc "Radio" remains as a "ghost" without any contents and you can not remove it in the ordinary way. It seems ETHOS Suite crashes more often after quitting if this "ghost" is present. Maybe this is a hint!?

**Comment by spoke2570** [comment]
BTW this is still happening on 1.6.2. Some time passes 30 Sec to like 1.5 min and the "suite quit unexpectedly" widow pops up.

**Comment by mawzthefinn** [comment]
Still present in Suite 1.7.0?

**Comment by PeterGraz** [comment]
I can't remember any crashes lately!

**Comment by spoke2570** [comment]
Yep. Actually more frequently now. I close the app by any approved method, and some seconds later I get the “Suite Closed Unexpectedly” popup from my OS. I’ll tack up a recent crash report when I get home.

**Comment by spoke2570** [comment]
Most recent crash report>>

[1.7_crash_report..txt](https://github.com/user-attachments/files/23962171/1.7_crash_report.txt)

**Comment by spoke2570** [comment]
Here is the report from my OS and the log files from FrSky Suite 2.0.1-RC1 from the same instance.

[from09102026.zip](https://github.com/user-attachments/files/32135713/from09102026.zip)
